src/components/App/Header/MarketMakerCountdown/MarketMakerCountdown.vue

```diff
@@ -0,0 +1,203 @@
+          {{ formattedDaysLeft }} {{ t('marketMakerCountdown.daysLeft') }}
+        </div>
+      </div>
+      <el-progress
+        class="countdown-info-progress"
+        :percentage="transactionsPercentage"
+        :show-text="false"
+        color="var(--s-color-theme-accent);"
+      />
+      <div class="countdown-info-counters">
+        <div class="counter-transactions">
+          <span class="counter-transactions__current">{{ accountMarketMakerInfo.count }}</span>
+          <span class="counter-transactions__total">&nbsp;/&nbsp;{{ total }}</span>
+          <span class="counter-transactions__unit">{{ t('marketMakerCountdown.txs') }}</span>
+        </div>
+        <div class="counter-volume">
+          <span>{{ t('marketMakerCountdown.volume') }}</span>
+          <span class="counter-volume__value"><formatted-amount :value="formattedVolume" integer-only /></span>
+        </div>
+      </div>
+      <s-divider class="countdown-info-divider" type="secondary" />
+      <div class="countdown-info-footer">
+        {{ t('marketMakerCountdown.description') }}&nbsp;
+        <a class="countdown-info-link" target="_blank" rel="nofollow noopener" :href="link">{{ t('learnMoreText') }}</a>
+      </div>
+    </div>
+    <countdown unit="MM" :percentage="transactionsPercentage" :count="accountMarketMakerInfo.count" />
+  </s-tooltip>
+</template>
+
+<script lang="ts">
+import { Component, Mixins, Watch } from 'vue-property-decorator';
+import { Getter, Action, State } from 'vuex-class';
+import { components, mixins } from '@soramitsu/soraneo-wallet-web';
+
+import Countdown from './Countdown.vue';
+import TranslationMixin from '@/components/mixins/TranslationMixin';
+
+import { Links, ZeroStringValue } from '@/consts';
+
+import type { AccountMarketMakerInfo } from '@sora-substrate/util';
+
+const REQUIRED_TX_COUNT = 500;
+const BLOCKS_IN_DAY = 14400;
+const BLOCKS_IN_MONTH = BLOCKS_IN_DAY * 30;
+
+@Component({
+  components: {
+    Countdown,
+    FormattedAmount: components.FormattedAmount,
+  },
+})
+export default class MarketMakerCountdown extends Mixins(mixins.NumberFormatterMixin, TranslationMixin) {
+  @Action('subscribeOnAccountMarketMakerInfo', { namespace: 'rewards' })
+  subscribeOnAccountMarketMakerInfo!: AsyncVoidFn;
+
+  @Action('unsubscribeAccountMarketMakerInfo', { namespace: 'rewards' })
+  unsubscribeAccountMarketMakerInfo!: AsyncVoidFn;
+
+  @Getter isLoggedIn!: boolean;
+  @Getter nodeIsConnected!: boolean;
+
+  @State((state) => state.rewards.accountMarketMakerInfo) accountMarketMakerInfo!: Nullable<AccountMarketMakerInfo>;
+  @State((state) => state.settings.blockNumber) blockNumber!: number;
+
+  @Watch('isLoggedIn')
+  @Watch('nodeIsConnected')
+  private async updateSubscriptions(value: boolean) {
+    if (value) {
+      await this.subscribeOnAccountMarketMakerInfo();
+    } else {
+      await this.unsubscribeAccountMarketMakerInfo();
```

Comment:
If component dies, is it enough to unsubscribe only here. Do we need to do the same in **beforeDestroy()** or it is unnecessary?

Reply:
Since the component is used in the header, it exists for the entire life cycle of the application. Therefore **beforeDestroy** will never be called (as it could be in the page component after navigation to another page)